ddl/partition.go

```diff
@@ -907,8 +898,15 @@ func onDropTablePartition(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		physicalTableIDs = removePartitionInfo(tblInfo, partNames)
 	}
 
-	rules := buildPlacementDropRules(job.SchemaID, tblInfo.ID, physicalTableIDs)
-	err = infosync.UpdatePlacementRules(nil, rules)
+	bundles := make([]*placement.Bundle, 0, len(physicalTableIDs))
+	for _, ID := range physicalTableIDs {
+		oldBundle, ok := d.infoHandle.Get().BundleByName(placement.GroupID(ID))
+		if ok && !oldBundle.IsEmpty() {
+			bundles = append(bundles, buildPlacementDropBundle(ID))
+		}
+	}
+
+	err = infosync.PutRuleBundles(nil, bundles)
```

Comment:
How about put `bundles==nil` judgement here instead of in `PutRuleBundles`?

Reply:
Drop table may share the logic, I think. I am OK to both sides.